File: backend/lib/order/order_item.rb
class OrderItem
  attr_reader :order, :product

  def initialize(order:, product:)
    @order = order
    @product = product
  end

  def total
    product.price
  end

  def process
    puts "Processing order item from #{product.name}"
    self.pick
    self.create_invoice
    self.ship
    puts "\n"
  end

  def pick
    puts "Picking physical item(#{product.name})"
  end

  def create_invoice
    puts 'Invoice created! All taxes must be paid'
  end

  def ship
    puts Shipping.label(@order)
  end
end

File: backend/proposal/core/order_item.py
class OrderItem:
    order = None
    product = None

    def __init__(self, order, product):
        self.order = order
        self.product = product

    def total(self):
        return 10
    
    # On this function, you can pass additionnal info to add on the shipping label
    def generate_shipping_label(self, additional_info = ""):
        return "Shipping Label\n{0}".format(additional_info)
